go/adbc/driver/bigquery/statement.go

```diff
@@ -0,0 +1,490 @@
+	switch key {
+	case OptionIntQueryMaxBillingTier:
+		st.queryConfig.MaxBillingTier = int(value)
+	case OptionIntQueryMaxBytesBilled:
+		st.queryConfig.MaxBytesBilled = value
+	case OptionIntQueryJobTimeout:
+		st.queryConfig.JobTimeout = time.Duration(value) * time.Millisecond
+	default:
+		return false, nil
+	}
+	return true, nil
+}
+
+func (st *statement) SetOptions(options map[string]string) error {
+	for k, v := range options {
+		v := v // copy into loop scope
+		handled, err := st.setQueryConfig(k, v)
+		if handled {
+			if err != nil {
+				return err
+			} else {
+				continue
+			}
+		}
+
+		switch k {
+		default:
+			return adbc.Error{
+				Code: adbc.StatusInvalidArgument,
+				Msg:  fmt.Sprintf("unknown database string type option `%s`", k),
+			}
+		}
+	}
+	return nil
+}
+
+func (st *statement) SetOption(key string, value string) error {
+	handled, err := st.setQueryConfig(key, value)
+	if handled {
+		if err != nil {
+			return err
+		} else {
+			return nil
+		}
+	}
+
+	return adbc.Error{
+		Code: adbc.StatusInvalidArgument,
+		Msg:  fmt.Sprintf("unknown statement string type option `%s`", key),
+	}
+}
+
+func (st *statement) SetOptionBytes(key string, value []byte) error {
+	return adbc.Error{
+		Code: adbc.StatusInvalidArgument,
+		Msg:  fmt.Sprintf("unknown statement bytes type option `%s`", key),
+	}
+}
+
+func (st *statement) SetOptionInt(key string, value int64) error {
+	handled, err := st.setQueryConfigInt(key, value)
+	if handled {
+		if err != nil {
+			return err
+		} else {
+			return nil
+		}
+	}
+
+	return adbc.Error{
+		Code: adbc.StatusInvalidArgument,
+		Msg:  fmt.Sprintf("unknown statement int type option `%s`", key),
+	}
+}
+
+func (st *statement) SetOptionDouble(key string, value float64) error {
+	return adbc.Error{
+		Code: adbc.StatusInvalidArgument,
+		Msg:  fmt.Sprintf("unknown statement double type option `%s`", key),
+	}
+}
+
+// SetSqlQuery sets the query string to be executed.
+//
+// The query can then be executed with any of the Execute methods.
+// For queries expected to be executed repeatedly, Prepare should be
+// called before execution.
+func (st *statement) SetSqlQuery(query string) error {
+	st.query = query
+	return nil
+}
+
+// ExecuteQuery executes the current query or prepared statement
+// and returns a RecordReader for the results along with the number
+// of rows affected if known, otherwise it will be -1.
+//
+// This invalidates any prior result sets on this statement.
+func (st *statement) ExecuteQuery(ctx context.Context) (array.RecordReader, int64, error) {
+	projectID, err := st.GetOption(OptionStringProjectID)
+	if err != nil {
+		return nil, -1, err
+	}
+	authType, err := st.GetOption(OptionStringAuthType)
+	if err != nil {
+		return nil, -1, err
+	}
+	credentials, err := st.GetOption(OptionStringCredentials)
+	if err != nil {
+		return nil, -1, err
+	}
+	client, err := newClient(ctx, projectID, authType, credentials)
+	if err != nil {
+		return nil, -1, err
+	}
```

Comment:
the client should have been created by the connection already. We should not create a new client for every call to execute. The client should be a member of the connection object